packages/flutter/lib/src/widgets/scroll_activity.dart

```diff
@@ -391,8 +396,17 @@ class ScrollDragController implements Drag {
     _lastDetails = details;
 
     // Build momentum only if dragging in the same direction.
-    if (_retainMomentum && velocity.sign == carriedVelocity!.sign)
+    // Check that the incomming velocity is more than
+    // a [momentumRetainVelocityThresholdFactor]'th of the carriedVelocity to
+    // avoid adding the carriedVelocity to the current velocity if the finger
+    // left the screen with a substantially smaller speed giving the effect of
+    // the scrollview carrying on in a much larger speed than the finger left with
+    if (_retainMomentum &&
+        velocity.sign == carriedVelocity!.sign &&
+        (carriedVelocity! == 0 || velocity.abs() > carriedVelocity!.abs() * momentumRetainVelocityThresholdFactor)
```

Comment:
There's no need to add the carriedVelocity if it is zero.

Reply:
Added each condition as a local